### PR TITLE
Additional Storage Replica parameters

### PR DESCRIPTION
**Pull request by robinharwood**
Updating the Storage Replica module with new parameters.

### CODE REVIEW

**Comment by robinharwood** [comment]
Hi @michaeltlombardi / @sdwheeler,

As per email, I believe this PR is release ready if you would be kind enough to review and if appropriate sign-off/merge 😊

The main update was to introduce a couple of new parameters but as usual I’ve also tried to address the other usually quality checklist items.

Please let me know if you have any questions or if I need to apply any other updates.

Thanks Robin

File: docset/winserver2022-ps/storagereplica/New-SRGroup.md [hunk 1]
```diff
@@ -16,88 +16,64 @@ Creates a replication group.
 ## SYNTAX
 
 ```
-New-SRGroup [[-ComputerName] <String>] [-Name] <String> [-VolumeName] <String[]> [-LogVolumeName] <String>
- [[-LogSizeInBytes] <UInt64>] [[-Description] <String>] [-EnableConsistencyGroups] [-EnableEncryption] [-Force]
- [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [<CommonParameters>]
+New-SRGroup [[-ComputerName] <String>] [-Name] <String> [-VolumeName] <String[]> [-LogVolumeName]
+<String> [[-LogSizeInBytes] <UInt64>] [[-Description] <String>] [-EnableConsistencyGroups]
+ [-EnableEncryption] [-EnableCompression] [-Force] [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>] [-AsJob] [<CommonParameters>]
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/New-SRGroup.md:22]: This was not indented properly. Also, don't split the type from the parameter.
```suggestion
New-SRGroup [[-ComputerName] <String>] [-Name] <String> [-VolumeName] <String[]>
 [-LogVolumeName] <String> [[-LogSizeInBytes] <UInt64>] [[-Description] <String>]
 [-EnableConsistencyGroups] [-EnableEncryption] [-EnableCompression] [-Force]
 [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [<CommonParameters>]
```

File: docset/winserver2022-ps/storagereplica/New-SRGroup.md [hunk 2]
```diff
@@ -16,88 +16,64 @@ Creates a replication group.
...
+
+The `New-SRGroup` cmdlet creates a replication group. A replication group contains one or more
+data volumes and an associated log volume. A replication group is the container for replication.
+
+> [!NOTE]
+> The **Compression** parameter is only available in
+> [Azure Stack HCI version 22H2](https://azure.microsoft.com/en-us/updates/public-preview-azure-stack-hci-version-22h2/) or later,
+> and Windows Server Datacenter: Azure Edition beginning with the 2022-09 Cumulative Update for Microsoft
+> server operating system version 21H2 for x64-based Systems
+> ([KB5017381](https://support.microsoft.com/help/5017381)).
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/New-SRGroup.md:35]: Remove the locale from all Microsoft URLs
```suggestion
> [!NOTE]
> The **Compression** parameter is only available in
> [Azure Stack HCI version 22H2](https://azure.microsoft.com/updates/public-preview-azure-stack-hci-version-22h2/) or later,
> and Windows Server Datacenter: Azure Edition beginning with the 2022-09 Cumulative Update for Microsoft
> server operating system version 21H2 for x64-based Systems
> ([KB5017381](https://support.microsoft.com/help/5017381)).
```

File: docset/winserver2022-ps/storagereplica/New-SRGroup.md [hunk 3]
```diff
@@ -159,10 +139,31 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -EnableCompression
+
+Indicates that this Storage Replica group should use SMB compression for data transfer. This
+parameter only applies to
+[Azure Stack HCI version 22H2](https://azure.microsoft.com/en-us/updates/public-preview-azure-stack-hci-version-22h2/)
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/New-SRGroup.md:146]:
```suggestion
[Azure Stack HCI version 22H2](https://azure.microsoft.com/updates/public-preview-azure-stack-hci-version-22h2/)
```

File: docset/winserver2022-ps/storagereplica/New-SRPartnership.md [hunk 4]
```diff
@@ -16,119 +16,94 @@ Creates a replication partnership between two replication groups.
...
-This cmdlet can create the complete replication topology.
-It can also tie together separately created replication groups.
+
+The `New-SRPartnership` cmdlet creates a replication partnership between two new or existing
+replication groups. This cmdlet can create the complete replication topology. It can also tie
+together separately created replication groups.
+
+> [!NOTE]
+> The **Compression** parameter is only available in
+> [Azure Stack HCI version 22H2](https://azure.microsoft.com/en-us/updates/public-preview-azure-stack-hci-version-22h2/) or later,
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/New-SRPartnership.md:48]:
```suggestion
> [Azure Stack HCI version 22H2](https://azure.microsoft.com/updates/public-preview-azure-stack-hci-version-22h2/) or later,
```

File: docset/winserver2022-ps/storagereplica/New-SRPartnership.md [hunk 5]
```diff
@@ -278,11 +259,33 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -EnableCompression
+
+Indicates that the connections in this partnership should use SMB compression for data transfer.
+This parameter only applies to
+[Azure Stack HCI version 22H2](https://azure.microsoft.com/en-us/updates/public-preview-azure-stack-hci-version-22h2/)
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/New-SRPartnership.md:266]:
```suggestion
[Azure Stack HCI version 22H2](https://azure.microsoft.com/updates/public-preview-azure-stack-hci-version-22h2/)
```

File: docset/winserver2022-ps/storagereplica/New-SRPartnership.md [hunk 6]
```diff
@@ -16,119 +16,94 @@ Creates a replication partnership between two replication groups.
...
 ```
-New-SRPartnership [[-SourceComputerName] <String>] [-SourceRGName] <String> [-DestinationComputerName] <String>
- [-DestinationRGName] <String> [[-ReplicationMode] <ReplicationMode>] [-PreventReplication] [-Seeded]
- [[-AsyncRPO] <UInt32>] [-EnableEncryption] [-Force] [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>]
- [-AsJob] [<CommonParameters>]
+New-SRPartnership [[-SourceComputerName] <String>] [-SourceRGName] <String>
+ [-DestinationComputerName] <String> [-DestinationRGName] <String> [[-ReplicationMode]
+<ReplicationMode>] [-PreventReplication] [-Seeded] [[-AsyncRPO] <UInt32>] [-EnableEncryption]
+ [-EnableCompression] [-Force] [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob]
+ [<CommonParameters>]
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/New-SRPartnership.md:25]:
```suggestion
New-SRPartnership [[-SourceComputerName] <String>] [-SourceRGName] <String>
 [-DestinationComputerName] <String> [-DestinationRGName] <String>
 [[-ReplicationMode] <ReplicationMode>] [-PreventReplication] [-Seeded] [[-AsyncRPO] <UInt32>]
 [-EnableEncryption] [-EnableCompression] [-Force] [-CimSession <CimSession[]>]
 [-ThrottleLimit <Int32>] [-AsJob] [<CommonParameters>]
```

File: docset/winserver2022-ps/storagereplica/Set-SRGroup.md [hunk 7]
```diff
@@ -16,65 +16,83 @@ Modifies settings of a replication group.
...
 ## DESCRIPTION
-The **Set-SRGroup** cmdlet modifies settings of an existing replication group.
+
+The `Set-SRGroup` cmdlet modifies settings of an existing replication group.
 A replication group contains one or more data volumes and an associated log volume.
 A replication group is the container for replication.
 
+> [!NOTE]
+> The **Compression** parameter is only available in
+> [Azure Stack HCI version 22H2](https://azure.microsoft.com/en-us/updates/public-preview-azure-stack-hci-version-22h2/) or later,
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/Set-SRGroup.md:51]:
```suggestion
> [Azure Stack HCI version 22H2](https://azure.microsoft.com/updates/public-preview-azure-stack-hci-version-22h2/) or later,
```

File: docset/winserver2022-ps/storagereplica/Set-SRGroup.md [hunk 8]
```diff
@@ -135,9 +158,31 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -Compression
+
+Indicates that this Storage Replica group should use SMB compression for data transfer. This
+parameter only applies to
+[Azure Stack HCI version 22H2](https://azure.microsoft.com/en-us/updates/public-preview-azure-stack-hci-version-22h2/)
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/Set-SRGroup.md:165]:
```suggestion
[Azure Stack HCI version 22H2](https://azure.microsoft.com/updates/public-preview-azure-stack-hci-version-22h2/)
```

File: docset/winserver2022-ps/storagereplica/Set-SRPartnership.md [hunk 9]
```diff
@@ -16,168 +16,137 @@ Modifies a replication partnership between two replication groups.
...
+
 ```
 Set-SRPartnership [[-SourceComputerName] <String>] [-SourceRGName] <String>
- [-SourceAddVolumePartnership] <String[]> [-DestinationComputerName] <String> [-DestinationRGName] <String>
- [-DestinationAddVolumePartnership] <String[]> [-Seeded] [-Force] [-CimSession <CimSession[]>]
- [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
+ [-SourceAddVolumePartnership] <String[]> [-DestinationComputerName] <String> [-DestinationRGName]
+<String> [-DestinationAddVolumePartnership] <String[]> [-Seeded] [-Force]
+ [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm]
+ [<CommonParameters>]
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/Set-SRPartnership.md:25]:
```suggestion
 [-SourceAddVolumePartnership] <String[]> [-DestinationComputerName] <String>
 [-DestinationRGName] <String> [-DestinationAddVolumePartnership] <String[]> [-Seeded] [-Force]
 [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm]
 [<CommonParameters>]
```

File: docset/winserver2022-ps/storagereplica/Set-SRPartnership.md [hunk 10]
```diff
@@ -16,168 +16,137 @@ Modifies a replication partnership between two replication groups.
...
 ```
-Set-SRPartnership [[-SourceComputerName] <String>] [-SourceRGName] <String> [-DestinationComputerName] <String>
- [-DestinationRGName] <String> [[-ReplicationMode] <ReplicationMode>] [[-LogSizeInBytes] <UInt64>]
- [[-AsyncRPO] <UInt32>] [[-Encryption] <Boolean>] [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>]
- [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
+Set-SRPartnership [[-SourceComputerName] <String>] [-SourceRGName] <String>
+ [-DestinationComputerName] <String> [-DestinationRGName] <String> [[-ReplicationMode]
+<ReplicationMode>] [[-LogSizeInBytes] <UInt64>] [[-AsyncRPO] <UInt32>] [[-Encryption] <Boolean>]
+ [[-Compression] <Boolean>] [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf]
+ [-Confirm] [<CommonParameters>]
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/Set-SRPartnership.md:35]:
```suggestion
Set-SRPartnership [[-SourceComputerName] <String>] [-SourceRGName] <String>
 [-DestinationComputerName] <String> [-DestinationRGName] <String>
 [[-ReplicationMode] <ReplicationMode>] [[-LogSizeInBytes] <UInt64>] [[-AsyncRPO] <UInt32>]
 [[-Encryption] <Boolean>] [[-Compression] <Boolean>] [-CimSession <CimSession[]>]
 [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
```

File: docset/winserver2022-ps/storagereplica/Set-SRPartnership.md [hunk 11]
```diff
@@ -16,168 +16,137 @@ Modifies a replication partnership between two replication groups.
...
 
 ### ModifyPartnership
+
 ```
-Set-SRPartnership [-SourceRGName] <String> [-DestinationComputerName] <String> [-DestinationRGName] <String>
- [-Force] [[-NewSourceComputerName] <String>] [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob]
- [-WhatIf] [-Confirm] [<CommonParameters>]
+Set-SRPartnership [-SourceRGName] <String> [-DestinationComputerName] <String> [-DestinationRGName]
+<String> [-Force] [[-NewSourceComputerName] <String>] [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/Set-SRPartnership.md:43]:
```suggestion
Set-SRPartnership [-SourceRGName] <String> [-DestinationComputerName] <String>
 [-DestinationRGName] <String> [-Force] [[-NewSourceComputerName] <String>]
 [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm]
 [<CommonParameters>]
```

File: docset/winserver2022-ps/storagereplica/Set-SRPartnership.md [hunk 12]
```diff
@@ -16,168 +16,137 @@ Modifies a replication partnership between two replication groups.
...
-You can use this cmdlet to add replicated volumes.
-You can also change the direction of replication which makes a source volume into a destination volume.
+
+The `Set-SRPartnership` cmdlet modifies a replication partnership between two existing replication
+groups. You can use this cmdlet to add replicated volumes. You can also change the direction of
+replication which makes a source volume into a destination volume.
+
+> [!NOTE]
+> The **Compression** parameter is only available in
+> [Azure Stack HCI version 22H2](https://azure.microsoft.com/en-us/updates/public-preview-azure-stack-hci-version-22h2/) or later,
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/Set-SRPartnership.md:54]:
```suggestion
> [Azure Stack HCI version 22H2](https://azure.microsoft.com/updates/public-preview-azure-stack-hci-version-22h2/) or later,
```

File: docset/winserver2022-ps/storagereplica/Set-SRPartnership.md [hunk 13]
```diff
@@ -231,7 +201,29 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -Compression
+
+Indicates that the connections in this partnership should use SMB compression for data transfer.
+This parameter only applies to
+[Azure Stack HCI version 22H2](https://azure.microsoft.com/en-us/updates/public-preview-azure-stack-hci-version-22h2/)
```
Comment (sdwheeler) [inline docset/winserver2022-ps/storagereplica/Set-SRPartnership.md:208]:
```suggestion
[Azure Stack HCI version 22H2](https://azure.microsoft.com/updates/public-preview-azure-stack-hci-version-22h2/)
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]
See my suggestions

**Comment by robinharwood** [comment]
Hey @sdwheeler,

Great review. I've accepted all your changes. Let me know if there is anything else?

Thanks

Robin

**Review by sdwheeler — APPROVED** [review]
LGTM

**Pull request merged**